scripts/functional-tests.sh

```diff
@@ -8,6 +15,14 @@ export SCRIPTDIR=$(dirname "$0")
 # build a prompt string that includes the time, source file, line number, and function name
 export PS4='+$(date +"%Y-%m-%d %T") ${BASH_VERSION}:${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
+if [ -n "$TEST_DIR" ]; then
```

Comment:
Please use double brackets when possible, this will help us reduce potential errors since pathname expansion or word splitting takes place between `[[` and  `]]`

```suggestion
if [[ -n "${TEST_DIR}" ]]; then
```

Reply:
It's just something I feel we should adhere for consistency in order to prevent bugs. But you're welcome to leave this as is, since it's currently passing tests anyway.